qiskit/converters/circuit_to_instruction.py

```diff
@@ -84,7 +84,7 @@ def circuit_to_instruction(circuit, parameter_map=None, equivalence_library=None
             "Circuits with internal variables cannot yet be converted to instructions."
             " You may be able to use `QuantumCircuit.compose` to inline this circuit into another."
         )
-    if CONTROL_FLOW_OP_NAMES.intersection(circuit.count_ops()):
+    if CONTROL_FLOW_OP_NAMES.intersection(op for op in circuit.count_ops() if op != "box"):
```

Comment:
I don't actually know the consequences of this, so I'm half expecting a qiskit core member to tell me to implement this PR in some other way.

Reply:
I would probably just do:

```suggestion
    op_counts = circuit.count_ops()
    op_counts.pop("box", None)
    if CONTROL_FLOW_OP_NAMES.intersection(op_counts):
```
So you avoid iterating over the dictionary. There might be a better way than that though.